### DISCUSSION TITLE
Generate pocketbase types without starting server?

### DISCUSSION THREAD

**Discussion by einarpersson** [pocketbase/pocketbase#4005] [Q&A] [answered]
When running pocketbase (i.e. pocketbase serve) the ambient types are saved to /pb_data.

Is it possible to output this without starting the server?

**Comment by ganigeorgiev** [comment]
> Is it possible to output this without starting the server?

What would be the use case for it?

You can register a dummy [console command](https://pocketbase.io/docs/js-console-commands/) and the JSVM types refresh method should get triggered by the `OnAfterBootstrap` hook on its own.

**Reply by einarpersson** [reply]
I write the hooks in typescript and transpile it (locally in dev, and in github workflow). For this step I need the types to be there. But I rather not track it in git, since it adds ~20k LOC. So then I want to be able to generate them easily.

**Reply by ganigeorgiev** [reply, marked as answer]
Any custom command or even running `./pocketbase migrate up` will generate them.
